src/Microsoft.Extensions.Options/IConfigureNamedOptions.cs

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace Microsoft.Extensions.Options
+{
+
+    /// <summary>
+    /// Represents something that configures the TOptions type.
+    /// </summary>
+    /// <typeparam name="TOptions"></typeparam>
+    public interface IConfigureNamedOptions<in TOptions> where TOptions : class
```

Comment:
It's not just sugar, but it requires to change the interface `IConfigureNamedOptions` and replace `string Name` with `string[] Names` (and change the realization for `Configure` method a bit).